cmake/HPX_AddModule.cmake

```diff
@@ -33,30 +33,34 @@ function(add_hpx_module name)
     ${HPX_WITH_TESTS} ADVANCED
     CATEGORY "Modules")
 
-  if(${name}_DEPRECATION_WARNINGS)
-    hpx_option(HPX_${name_upper}_WITH_DEPRECATION_WARNINGS
-      BOOL
-      "Enable warnings for deprecated facilities. (default: ${HPX_WITH_DEPRECATION_WARNINGS})"
-      ${HPX_WITH_DEPRECATION_WARNINGS} ADVANCED
-      CATEGORY "Modules")
-    if(HPX_${name_upper}_WITH_DEPRECATION_WARNINGS)
-      hpx_add_config_define_namespace(
-          DEFINE HPX_${name_upper}_HAVE_DEPRECATION_WARNINGS
-          NAMESPACE ${name_upper})
-    endif()
+  set(_deprecation_warnings_default OFF)
+  if(${HPX_WITH_DEPRECATION_WARNINGS} AND ${name}_DEPRECATION_WARNINGS)
+    set(_deprecation_warnings_default ON)
+  endif()
+  hpx_option(HPX_${name_upper}_WITH_DEPRECATION_WARNINGS
+    BOOL
+    "Enable warnings for deprecated facilities. (default: ${HPX_WITH_DEPRECATION_WARNINGS})"
+    ${_deprecation_warnings_default} ADVANCED
+    CATEGORY "Modules")
+  if(${HPX_${name_upper}_WITH_DEPRECATION_WARNINGS})
+    hpx_add_config_define_namespace(
+      DEFINE HPX_${name_upper}_HAVE_DEPRECATION_WARNINGS
+      NAMESPACE ${name_upper})
   endif()
 
-  if(${name}_COMPATIBILITY_HEADERS)
-    hpx_option(HPX_${name_upper}_WITH_COMPATIBILITY_HEADERS
-      BOOL
-      "Enable compatibility headers for old headers"
-      ON ADVANCED
-      CATEGORY "Modules")
-    if(HPX_${name_upper}_WITH_COMPATIBILITY_HEADERS)
-      hpx_add_config_define_namespace(
-          DEFINE HPX_${name_upper}_HAVE_COMPATIBILITY_HEADERS
-          NAMESPACE ${name_upper})
-    endif()
+  set(_compatibility_headers_default OFF)
+  if(${HPX_WITH_COMPATIBILITY_HEADERS} AND ${name}_COMPATIBILITY_HEADERS)
```

Comment:
I see the appeal of the global option but we may end up with compatibility headers that have been introduced in different versions for different modules. This puts all the modules on the same "cycle". We also end up with `HPX_WITH_MODULENAME_COMPATIBILITY_HEADERS` variables for modules that don't have any. I'd prefer not doing this, but don't mind if there's a strong case for this.

Reply:
This is a fair point. I wanted to have a global option I could use for a sensible default. I'll revert this particular change.

Reply:
That has been taken care of now.